pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -336,10 +329,38 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) (Results, err
 	return Results{
 		NewNodeClaims: s.newNodeClaims,
 		ExistingNodes: s.existingNodes,
-		PodErrors:     podErrors,
+		PodErrors: lo.MapEntries(podErrors, func(id types.UID, err error) (*corev1.Pod, error) {
```

Comment:
I'm thinking we should update this to use UID as well - I don't think it needs to be scoped to this PR, but we could open an issue.

Reply:
It's currently not possible based on how we handle pod errors in disruption -- agreed that for now it's fine -- and honestly, I don't think it hurts a ton to use the pod as the actual key

Reply:
Probably not, we just have to be careful if we're every making copies.